test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs

```diff
@@ -248,123 +262,108 @@ private void CopyLatestCliToTestDirectory(string destinationDir)
 
         private void UpdateCliWithLatestNuGetAssemblies(string cliDirectory)
         {
-            var nupkgsDirectory = DotnetCliUtil.GetNupkgDirectoryInRepo();
-
-            var pathToPackNupkg = FindMostRecentNupkg(nupkgsDirectory, "NuGet.Build.Tasks.Pack");
-
-            var nupkgsToCopy = new List<string> { "NuGet.Build.Tasks", "NuGet.Versioning", "NuGet.Protocol", "NuGet.ProjectModel", "NuGet.Packaging", "NuGet.LibraryModel", "NuGet.Frameworks", "NuGet.DependencyResolver.Core", "NuGet.Configuration", "NuGet.Common", "NuGet.Commands", "NuGet.CommandLine.XPlat", "NuGet.Credentials" };
-
+            var artifactsDirectory = DotnetCliUtil.GetArtifactsDirectoryInRepo();
             var pathToSdkInCli = Path.Combine(
                     Directory.GetDirectories(Path.Combine(cliDirectory, "sdk"))
                         .First());
+            const string configuration =
+#if DEBUG
+                "Debug";
+#else
+                "Release";
+#endif
+            CopyPackSdkArtifacts(artifactsDirectory, pathToSdkInCli, configuration);
+            CopyRestoreArtifacts(artifactsDirectory, pathToSdkInCli, configuration);
+        }
 
-            using (var nupkg = new PackageArchiveReader(pathToPackNupkg))
-            {
-                var pathToPackSdk = Path.Combine(pathToSdkInCli, "Sdks", "NuGet.Build.Tasks.Pack");
-                var files = nupkg.GetFiles()
-                .Where(fileName => fileName.StartsWith("Desktop")
-                                || fileName.StartsWith("CoreCLR")
-                                || fileName.StartsWith("build")
-                                || fileName.StartsWith("buildCrossTargeting"));
-
-                DeleteDirectory(pathToPackSdk);
-                CopyNupkgFilesToTarget(nupkg, pathToPackSdk, files);
-            }
-
+        private void CopyRestoreArtifacts(string artifactsDirectory, string pathToSdkInCli, string configuration)
+        {
+            const string restoreProjectName = "NuGet.Build.Tasks";
+            const string restoreTargetsName = "NuGet.targets";
+            var sdkDependencies = new List<string> { restoreProjectName, "NuGet.Versioning", "NuGet.Protocol", "NuGet.ProjectModel", "NuGet.Packaging", "NuGet.LibraryModel", "NuGet.Frameworks", "NuGet.DependencyResolver.Core", "NuGet.Configuration", "NuGet.Common", "NuGet.Commands", "NuGet.CommandLine.XPlat", "NuGet.Credentials" };
 
-            foreach (var nupkgName in nupkgsToCopy) {
-                using (var nupkg = new PackageArchiveReader(FindMostRecentNupkg(nupkgsDirectory, nupkgName)))
+            // Copy rest of the NuGet assemblies.
+            foreach (var projectName in sdkDependencies)
+            {
+                var projectArtifactsFolder = new DirectoryInfo(Path.Combine(artifactsDirectory, projectName, "16.0", "bin", configuration));
```

Comment:
I've combined the 2 reference to "16.0" into a constant, but beyond that other approaches that I can think of are equally error prone. It requires a manual switch. And since the only place where we do things like this is test code, not sure if it's moving the work to build code. 

Open to ideas if you have any.